src/plugins/discover/public/application/main/components/layout/discover_main_content.tsx

```diff
@@ -60,13 +60,17 @@ export const DiscoverMainContent = ({
   panelsToggle,
   isChartAvailable,
 }: DiscoverMainContentProps) => {
-  const { trackUiMetric, dataVisualizer: dataVisualizerService } = useDiscoverServices();
+  const { trackUiMetric } = useDiscoverServices();
   const isEsqlMode = useIsEsqlMode();
-  const shouldShowViewModeToggle = dataVisualizerService !== undefined;
 
   const setDiscoverViewMode = useCallback(
     (mode: VIEW_MODE) => {
-      stateContainer.appState.update({ viewMode: mode });
+      stateContainer.appState.update(
+        // Resetting to default viewMode: undefined if document level
+        // else change for viewMode from undefined to DOCUMENT_LEVEL will triger unsaved changes badge
```

Comment:
If I save a search with Field Statistics tab selected and then switch to Documents, I don't see the badge any more although it was there before the PR.

@kertal @davismcphee What behaviour is preferable?

Reply:
I'd say this badge should be displayed since the modified state is different to the persisted one, and if it's saved again, Documents instead of Field statistics are displayed when the saved search is loaded (and on a Dashboard).

Reply:
For data view mode the badge should be displayed when switching tabs since the selected tab is persisted to the saved search, so it represents a change in the state.

For ES|QL mode it depends on the intended behaviour. If we will support persisting the tab to the saved search, it _should_ should the badge. If we choose not to support persisting the tab, it _shouldn't_ show the badge. My personal preference is for consistency between modes, but I don't feel strongly about it so I'd defer to @stratoula on that one.

Reply:
I agree with you Davis, if we don't persist the change then is not considered a state change. I will be completely honest here I am in the middle. On one hand I like consistency on the other hand I find it very weird to persist the fields statistics tab. Was it a feature request? I feel we should do something different here. But is def a business desicion.

My proposal is:

- merge this PR without the persistence. We are going to discuss it on Tue and if we decide in favor of it we can have a follow up PR
- wait till we have our sync on Tue, decide and merge after

It really depends on you @qn895 , I know how painful can be to keep a PR open for so long. I dont think also that is the end of the world if we merge the PR now and adjust in a week or so.

Reply:
I'm ok with merging with either behaviour since we're going to follow up on it after anyway, as long as the unsaved changes badge behaves correctly based on my previous comment.

I'm not sure if it was a feature request since it's been around longer than I have, but one reason I can think of for why it might work that way is that it's currently the only way to embed field statistics for a saved search within a Dashboard panel.

Reply:
@davismcphee  @stratoula Thanks all for the reviews. To recap from our ES|QL sync, for this PR I will:
- Keep the state (documents vs field stats mode) saved in both non-ESQL and ESQL mode for consistency. So loading the saved search in the dashboard will persist the state when user saves the search. In the future, we can expand on the saved search experience to allow for finer grain control.
- Show the `unsaved changes` badge whenever the tab is changed, and update the functional tests to reflect that

Reply:
Yes that sounds right to me, thanks!